docs/checks.md

```diff
@@ -0,0 +1,53 @@
+# Checks (Beta)
+Checks are data integrity tests that are defined in `build.yml`
```

Comment:
recommend a little more color:
- one sentence on the vision
- example output
- 1-3 sentences on a practical guide when/why to use checks and when not to
- known limitations and already-regretted design decisions :-)
    - scaling: no progress indicators
    - scaling: what happens if a node's data doesn't fit in a panda dataframe
    - future: convenience syntax for NOT-checks
    - future: convenience syntax for per-row checks
- checks.yml and when to use it (e.g. large checks-code and also for reuse across buildfiles)

note that checks are one of the first major features where build.yml recovery is a big deal, and ideally, this would be noted in limitations.